src/init/features/apphosting/repo.ts

```diff
@@ -135,9 +135,9 @@ async function promptRepositoryUri(
   const remoteUriToConnection: Record<string, gcb.Connection> = {};
   for (const conn of connections) {
     const { location, id } = parseConnectionName(conn.name)!;
-    const resp = await gcb.fetchLinkableRepositories(projectId, location, id);
-    if (resp.repositories && resp.repositories.length > 0) {
-      for (const repo of resp.repositories) {
+    const repos = await gcb.fetchLinkableRepositories(projectId, location, id);
```

Comment:
We're probably gonna run into some merge conflicts here, I reworked the logic here as well (https://github.com/firebase/firebase-tools/pull/6764/files#diff-a82ad0512cae837b1f50b301e857637957d3b8dc5fb4d2c8095b7a2853857179)